### PR TITLE
Added type convertion on expression projection

### PR DESCRIPTION
**Pull request by HRasch**
Expression Projection with different types #2538

### CODE REVIEW

**Comment by HRasch** [comment]
Hm, first of all, what do you mean with "not using ToType as discussed."
You just said
> cc @BlaiseD I don't know what's the right fix here, but we have a ToType helper.

so i changed the call of 
`
replacedExpression = Convert(replacedExpression, node.Type);
`
to 

`
replacedExpression = ExpressionFactory.ToType(replacedExpression, node.Type);
`
using the suggested convertion-method. (Imo, it would be nice if this method would be an extensions-method)

Sure, it would be much better if this call would also support the propertyMap, but that's another story.

On the other hand, I'm really struggled running the existing tests of automapper.
[Contributing.md](https://github.com/AutoMapper/AutoMapper/blob/master/CONTRIBUTING.md) said I should use psake.cmd, but even after upgrade my VS2017, installing .NET Core 1.1.6 SDK, having trouble because AutoMapper.UnitTests is definied to target net40, even if [support has been dropped with 4.2.0](https://github.com/AutoMapper/AutoMapper/issues/1078), using different versions of xunit (because of net40 support), xunit.runner.visualstudio (including a 2.3.0-beta3, even if 2.3.1 rtm is released), I'm not able to run the tests with the visual studio test explorer. Even running psake.cmd throws compile-errors on the existing tests: E.g.:
> No coercion operator is defined between types 'UnitTests.Bug.ExpressionMappingWithUseAsDataSource+Model' and 'UnitTests.Bug.ExpressionMappingWithUseAsDataSource+DTO+DTONested'

So, It's not hard for me to write my unit test, but i would feel much better if i could compile and test it :-)

File: src/AutoMapper/Mappers/ExpressionMapper.cs [hunk 1]
```diff
@@ -209,9 +210,18 @@ protected override Expression VisitMember(MemberExpression node)
                     replacedExpression = _parentMappingVisitor.Visit(node.Expression);
 
                 if (propertyMap.CustomExpression != null)
-                    return propertyMap.CustomExpression.ReplaceParameters(replacedExpression);
+				{
+					replacedExpression = propertyMap.CustomExpression.ReplaceParameters(replacedExpression);
 
-                Func<Expression, MemberInfo, Expression> getExpression = MakeMemberAccess;
+					if (!node.Type.IsAssignableFrom(replacedExpression.Type))
```
Comment (lbargaoanu) [inline src/AutoMapper/Mappers/ExpressionMapper.cs]: This check is not needed.

**Comment by lbargaoanu** [comment]
Yes, I guess I got distracted by the extra check :) I usually run the tests in VS, but "dotnet test" works too. You can try changing the order of frameworks in csproj. It can be a struggle to make it work, but it's possible. And yes, an extension method would be nice.

**Comment by HRasch** [comment]
Ok, removed the check now. I added it before, because it's exactly the check that MS do in it's MemberAssignment.Expressions.Bind-Method and caused the error before. But the internal check of quality should do (nearly) the same.

**Comment by lbargaoanu** [comment]
Maybe I'm missing smth, but the test runs just fine on master.

**Pull request closed without merging**